bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/values/ColorValue.java

```diff
@@ -120,7 +120,7 @@ public void update(Command command) throws IllegalArgumentException {
         }
     }
 
-    private static BigDecimal factor = new BigDecimal(2.5);
+    private static BigDecimal factor = new BigDecimal("2.55"); // string to not lose precision
```

Comment:
Would `new BigDecimal(2.55)` not work ? I am not sure to understand the need for a  string.

Could you move up this constant at the beginning of the class please.

Reply:
Because `new BigDecimal(2.55).toString()` yields `2.54999999999999982236431605997495353221893310546875`. In other words, `new BigDecimal(2.55).multiply(new BigDecimal(100)).intValue()` yields `254`, not `255`.

Yes, I'll move it to the top.